### DISCUSSION TITLE
Flexible online code editor

### DISCUSSION THREAD

**Discussion by jeswanth9999** [scaleracademy/open-source-projects#75] [Ideas]
Main focus here is to provide an easily integratable code editor which is fast in performance.
-- Any one integrating by themselves can add the path of compiler and editor picks it and start working. 
-- Can adjust the code editor as they want
-- If someone kept unnecessary for loop it should suggest we should use better option (something like this not exactly)
-- displaying time complexity or suggesting better time complexity of code. 
-- If someone practicing competitive programing they can  on a separate mode which having features like showing execution time, time complexity, suggest  questions related to it from internet.
-- There will be a note option where it stores in local if someone is using personal and in server if on web. Where people working on code can add some text for later understanding.
-- Reminder for certain projects/files loaded to get complete(If we add something for working project or class/page. When ever we open the code editor it will show things to complete).

Features:
-- Can sync the project with others to fix there issues.
-- Team collaboration where it shows who written particular code.
-- Can ask someone to fix the code if necessary so message or notification goes to them asking for a fix.
-- An option to add the  integrations (of course all coding editor provide) but we provide they can integrate or add from any languages.

**Comment by AgarwalPragy** [comment]
This one is too ambitious..

1. an Online editor will not be able to use a compiler that you've installed locally - browsers won't allow any js to access local files
2. displaying the time complexity, or suggesting a program with better time complexity is fundamentally an unsolvable problem (see Halting Problem).

It is easy to come up with a grand vision of an extremely complex project. 
The real challenge is to figure out what parts are essential, and what parts can actually be built by a team of 2-3 developers.

**Reply by jeswanth9999** [reply]
Hi,

1. We will be picking the path of the files and use it like user selecting the files from the local. This will be done during the installation itself. Where we ask for language and path for the files. Alternative for it is using webserver and picking the files from local.

2. My idea towards the time complexity for the competitive programming but not real project is we try to calculate based on the syntax of that language. We read the file to check the loops and variables to provide the time complexity.

Yes, I understand it can't be done with small team but as it is opensource may people who are interested will contribute. The whole purpose of this is for people to learn and providing the best solution for the users in some point of time.
